backend/pkg/modules/internal/differ/diffs_sender.go

```diff
@@ -0,0 +1,115 @@
+package differ
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+
+	"github.com/openclarity/apiclarity/api3/common"
+	"github.com/openclarity/apiclarity/api3/global"
+	"github.com/openclarity/apiclarity/api3/notifications"
+	"github.com/openclarity/apiclarity/backend/pkg/config"
+	"github.com/openclarity/apiclarity/backend/pkg/database"
+)
+
+// unit tests
+
+func (p *pluginDiffer) StartDiffsSender(ctx context.Context) {
+	// each period aggregate diifs per api and notify to notification server
+	// TODO configurable
+	log.Info("Starting diffs sender")
+	interval := viper.GetInt(config.DiffsSendIntervalMin)
+
+	ticker := time.NewTicker(time.Duration(interval) * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			if err := p.sendDiffsNotification(); err != nil {
+				log.Errorf("Failed to send diffs notification: %v", err)
+				// TODO do we want to keep the diffs in case of failure?
+			}
+			// clear diffs
+			p.clearDiffs()
```

Comment:
Doe this mean that at every cycle only new diffs are sent? Do you expect the mgmt to remember past diffs or you expect that older diffs just disappear?

Reply:
I will send the hash of the event (oldSpec+newSpec) to the management so they should know about already existing diffs.

Reply:
The management keeps all diffs in their own database, so no need to keep old diffs